combin/combin.go

```diff
@@ -0,0 +1,169 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package combin implements routines involving combinatorics (permutations,
+// combinations, etc.).
+package combin
+
+import "math"
+
+const (
+	badNegInput = "combin: negative input"
+	badSetSize  = "combin: n < k"
+	badInput    = "combin: wrong input slice length"
+)
+
+// Binomial returns the binomial coefficient of (n,k).
+//
+// The binomial coefficient is the number of unordered combinations of k elements
+// in a set that is n elements big. It is also commonly referred to as "n choose k".
+//
+//  n!/((n-k)!k!)
```

Comment:
I like the formulas to read as a part of the text. How about restructuring the comment as:
```
// Binomial returns the binomial coefficient of (n,k), also commonly referred to as "n choose k".
//
// The binomial coefficient C(n,k) is the number of unordered combinations of k elements
// in a set that is n elements big, and is defined as
//
//  C(n,k) := n!/((n-k)!k!).
//
// n and k must be non-negative with n >= k, otherwise Binomial will panic.
// No check is made for overflow.
```
Just a suggestion.